integration/rpcserver_test.go

```diff
@@ -94,10 +95,45 @@ func testGetBlockHash(r *rpctest.Harness, t *testing.T) {
 	}
 }
 
+func testBulkClient(r *rpctest.Harness, t *testing.T) {
+	// Create a new block connecting to the current tip.
+	generatedBlockHashes, err := r.Client.Generate(20)
+	if err != nil {
+		t.Fatalf("Unable to generate block: %v", err)
+	}
+
+	var futureBlockResults []rpcclient.FutureGetBlockResult
+	for _, hash := range generatedBlockHashes {
+		futureBlockResults = append(futureBlockResults, r.BatchClient.GetBlockAsync(hash))
+	}
+
+	err = r.BatchClient.Send()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+BLOCKRESULT:
+	for _, block := range futureBlockResults {
+		msgBlock, err := block.Receive()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		blockHash := msgBlock.Header.BlockHash()
+		for _, hash := range generatedBlockHashes {
+			if blockHash.IsEqual(hash) {
+				continue BLOCKRESULT
+			}
+		}
+		t.Fatalf("expected hash %s  to be in generated hash list", blockHash)
+	}
```

Comment:
Can we rewrite this using a helper function for checking if a received blockhash is part of the generated list? The goto statement makes it a bit hard to understand. Here's an example below (not tested):

```suggestion
    isKnownBlockHash := func(hash string) bool {
        for _, h := range generatedBlockHashes {
            if hash == h {
                return true
            }
        }

        return false
    }

    for _, block := range futureBlockResults {
        msgBlock, err := block.Receive()
        if err != nil {
	    t.Fatal(err)
        }

        blockHash := msgBlock.Header.BlockHash()
        if !isKnownBlockHash(blockHash) {
            t.Fatalf("expected hash %s  to be in generated hash list", blockHash)
        }
    }
```